service-api/module/Application/src/Controller/YotiController.php

```diff
@@ -56,22 +64,37 @@ public function findPostOfficeAction(): JsonModel
         $this->getResponse()->setStatusCode(Response::STATUS_CODE_200);
         return new JsonModel($branches);
     }
-    public function createSessionAction(array $sessionData): JsonModel
+    public function createSessionAction(): JsonModel
     {
         $uuid = $this->params()->fromRoute('uuid');
-        //@TODO authenticate if not using mock?
-        $result = $this->yotiService->createSession($sessionData);
-        //save sessionId back to caseData
-        if ($result["status"] < 400) {
-            $this->dataImportHandler->updateCaseData(
-                $uuid,
-                'sessionId',
-                'S',
-                $result["data"]["session_id"]
-            );
+
+        if (! $uuid) {
+            $this->getResponse()->setStatusCode(Response::STATUS_CODE_400);
+            return new JsonModel(['error' => 'Missing uuid']);
         }
-        $this->getResponse()->setStatusCode(Response::STATUS_CODE_204);
-        return new JsonModel(null);
+        $sessionUuid = strval(Uuid::uuid4());
+        $caseData = $this->dataQuery->getCaseByUUID($uuid);
+        $sessionData = $this->sessionConfig->build($caseData, $sessionUuid);
+
+        try {
+            $result = $this->yotiService->createSession($sessionData);
+        } catch (YotiException $e) {
+            $this->getResponse()->setStatusCode(Response::STATUS_CODE_400);
+            return new JsonModel(new Problem(
+                'Problem sending request',
+                extra: ['errors' => $e->getMessage()],
+            ));
+        } catch (YotiApiException $apiException) {
```

Comment:
What's the difference between YotiException and YotiApiException? They both seem to be thrown as a result of errors from the Yoti API, just from different methods?

I'm also not sure the response codes are right here: `YotiException` will be thrown if the PEM file is invalid, which should result in a 500 error (because the server's misconfigured), not a 400 error.

I feel like `YotiAuthException` would be a good start for a more descriptive class, maybe there are other common more-specific cases too?

Reply:
Think I was using YotiException for pem file/signing issues and then created YotiApiException as a way to differentiate any errors potentially thrown my Yoti itself. 

Maybe YotiAuthException does sound more descriptive to what it's going. Can condense this into one exception with a 500?

Reply:
I reckon these are three good exceptions to start with:

- YotiAuthException: for failures in the auth/signing process
- YotiClientException: for networking errors (like the Yoti server being unavailable)
- YotiException: for all other unexpected errors (like `/create-session` fails even though we think we gave it correct info)

However, all the handling routes are basically the same (return 500) so it might be worth making YotiAuthException and YotiClientException extend YotiException, so you can just do `catch (YotiException $e)` rather than having to handle all three of them separately.

Reply:
Sounds good, thanks Greg :)